### PR TITLE
Added support for download of all available subtitles on a video

### PR DESCRIPTION
**Pull request by qnorsten**
Added command to download all available subtitles for a video (--all-subtitles)
Added support to print all the subtitle urls when the get url parameter is used
Fixed so subtitle url and stream url get printed if -S and -g but not --force-subtitles parameter is used
Added support for downloading all subtitles and auto subfix them with language name for Urplay and Urskola (even when just one subtitle is downloaded)

### CODE REVIEW

File: lib/svtplay_dl/subtitle/__init__.py [hunk 1]
```diff
@@ -11,12 +11,13 @@
 
 
 class subtitle(object):
-    def __init__(self, options, subtype, url):
+    def __init__(self, options, subtype, url, subfix = ""):
```
Comment (spaam) [inline lib/svtplay_dl/subtitle/__init__.py]: you can write it like: subfix = None :)

File: lib/svtplay_dl/subtitle/__init__.py [hunk 2]
```diff
@@ -37,7 +38,10 @@ def download(self):
                 data = subdata.text.encode("utf-8")
             else:
                 data = subdata.text
-
+        
+        if(self.subfix != ""):
```
Comment (spaam) [inline lib/svtplay_dl/subtitle/__init__.py]: if self.subfix: 
No need for ( and )

File: lib/svtplay_dl/__init__.py [hunk 3]
```diff
@@ -218,12 +219,23 @@ def get_one_media(stream, options):
         log.info("No subtitles available")
         return
 
-    if options.subtitle and options.get_url and options.force_subtitle:
-        print(subs[0].url)
-        return
+    if options.subtitle and options.get_url:
+        if options.get_all_subtitles:
+            for sub in subs[:]:
```
Comment (spaam) [inline lib/svtplay_dl/__init__.py]: no need for [:].  for sub in subs  will traverse the whole list

File: lib/svtplay_dl/__init__.py [hunk 4]
```diff
@@ -218,12 +219,23 @@ def get_one_media(stream, options):
...
+        else:
+            print(subs[0].url)
+        if options.force_subtitle: 
+            return
+        
     if options.subtitle and options.output != "-" and not options.get_url:
         if subs:
-            subs[0].download()
+            if options.get_all_subtitles:
+                for sub in subs[:]:
```
Comment (spaam) [inline lib/svtplay_dl/__init__.py]: no need for [:].  for sub in subs  will traverse the whole list

**Comment by qnorsten** [comment]
Thanks for the comments spaam. Always good with feedback.

**Pull request merged**